File: UVa/UVa_11310.cpp
#include <bits/stdc++.h>
const int limN = 40;

long long ans[limN+1];

void precarga() {
    ans[0] = ans[1] = 1LL;
    ans[2] = 5LL;
    for(int i=3; i<=limN; i++) 
        ans[i] = ans[i-1] + 4LL*ans[i-2] + 2LL*ans[i-3];
}

int main() {
    precarga();
    int tc, a;

    for(scanf("%d", &tc); tc; tc--) {
        scanf("%d", &a);
        printf("%lld\n", ans[a]);
    }
}

File: UVa/UVa_11310.py
limN = 40
ans = [0 for i in range(40+1)]


def precarga():
    global limN, ans
    ans[0] = 1
    ans[1] = 1
    for i in range(2, limN+1):
        ans[i] = ans[i-1] + 4*ans[i-2]


def main():
    precarga()
    tc = int(input())
    for i in range(tc):
        a = int(input())
        print(ans[a])
